### ISSUE TITLE
Is it possible to change Diffie Hellman key's path?

### ISSUE THREAD

**Issue by spunszor** [streamlink/streamlink#1462]
Is it possible to change the DH key's (CAB/keyfile) path of a HLS stream?

The program autofills this and I couldn't find where to declare a new path because "/CAB/keyfile?t=VOD&r=140874&p=0" the program gives is actully not the right one. I have the right path, but i don't know how to change it.

Sorry if this is a dumb question, but I couldn't find the answer here and neither in the documentation.

**Comment by back-to** [comment]
```
#EXT-X-KEY:METHOD=AES-128,URI=".../CAB/keyfile?...",
```

do you mean updating this url?
with the current stat of streamlink it's not possible

---

but it could be easely added if the interest is there.

**Comment by back-to** [comment]
not sure if this feature should be added to the master branch,
but you can sideload this plugin and it should work for you locally

https://streamlink.github.io/cli.html#sideloading-plugins

https://raw.githubusercontent.com/back-to/plugins/master/plugins/hlskeyuri.py

---

the old url parts can be reused **scheme, netloc, path, query**

**example url**

```py
'http://www.cwi.nl:80/%7Eguido/Python.html'

scheme='http', netloc='www.cwi.nl:80', path='/%7Eguido/Python.html', query='',
```

https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

**example streamlink**

```
${scheme} ${netloc} ${path} ${query}
streamlink hlskeyuri://URL --hls-key-uri '${scheme}${netloc}${path}${query}'
```

like this

```
Replace the broken part, like:
streamlink hlskeyuri://URL --hls-key-uri 'https://${netloc}${path}${query}'
```

**Comment by gravyboat** [comment]
Fixed via #2139
